SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialAuthorityTest/SpatialAuthorityTestActorComponent.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Components/SceneComponent.h"
+#include "SpatialAuthorityTestActorComponent.generated.h"
+
+class ASpatialFunctionalTest;
+class ASpatialFunctionalTestFlowController;
+
+UCLASS()
+class SPATIALGDKFUNCTIONALTESTS_API USpatialAuthorityTestActorComponent : public USceneComponent
+{
+	GENERATED_BODY()
+	
+public:	
+	USpatialAuthorityTestActorComponent();
+
+	void GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const override;
+
+	virtual void OnAuthorityGained() override;
+
+	virtual void OnAuthorityLost() override;
+
+	virtual void TickComponent(float DeltaTime, enum ELevelTick TickType, FActorComponentTickFunction *ThisTickFunction);
+	
+	UPROPERTY(Replicated)
+	int ReplicatedAuthorityOnBeginPlay = 0;
+
+	int AuthorityOnBeginPlay = 0;
```

Comment:
nit: can we slightly rename this variables to `AuthWorkerIdOnBeginPlay` `AuthWorkerIdOnTick` `ReplicatedAuthWorkerIdOnBeginPlay`